src/java.base/share/classes/java/lang/LazyConstant.java

```diff
@@ -150,46 +140,41 @@
  * <p>
  * The invocation of the computing function and the resulting initialization of
  * the constant {@linkplain java.util.concurrent##MemoryVisibility <em>happens-before</em>}
- * the initialized constant is read. Hence, the initialized constant, including any
- * {@code final} fields of any newly created objects, is safely published.
+ * the initialized constant's content is read. Hence, the initialized constant's content,
+ * including any {@code final} fields of any newly created objects, is safely published.
  * <p>
  * Thread interruption does not cancel the initialization of a lazy constant. In other
  * words, if the computing thread is interrupted, {@code LazyConstant::get} doesn't clear
- * the interrupted thread’s status, nor does it throw an InterruptedException.
+ * the interrupted thread’s status, nor does it throw an {@linkplain InterruptedException}.
  * <p>
  * If the computing function blocks indefinitely, other threads operating on this
  * lazy constant may block indefinitely; no timeouts or cancellations are provided.
  *
  * <h2 id="performance">Performance</h2>
- * A lazy constant can never change after it has been initialized. Therefore,
- * a JVM implementation may, for an initialized lazy constant, elide all future reads
- * of that lazy constant and instead directly use any constant that it has previously
- * observed. We call this optimization <em>constant folding</em>. This is only possible if
- * the reference to the lazy constant is a VM constant (e.g. in cases where
- * the lazy constant itself is stored in a {@code static final} field) or forms
- * a trusted chain to such a VM constant via one or more trusted fields (i.e., any
- * combination of static final fields, {@linkplain Record record} fields, or
- * final instance fields in hidden classes).
+ * The contents of lazy constant can never change after it has been initialized. Therefore,
+ * a JVM implementation may, for an initialized lazy constant, elide all future reads of
+ * that lazy constant's contents and instead use the contents that has been previously
+ * observed. We call this optimization <em>constant folding</em>. This is only possible
```

Comment:
This is still not ok.

First, there seems to be some typo ("if the reference to the lazy constant in a static final field or...").

Secondly, trusted chain _to_ such field. I think it's the other way around, the trusted chain starts at a static final field, and then eventually gets to the lazy constant, possibly traversing one or more trusted fields.